addons/account/models/company_onboarding.py

```diff
@@ -0,0 +1,163 @@
+            # The company data onboarding is marked as done if street1 is filled
+            record.onboarding_company_done = all_done or bool(self.street)
+            # The invoice onboarding is marked as done if logo is filled
+            # and different from the default one default
+            record.onboarding_invoice_done = all_done or(
+                        bool(self.logo) and self.logo != self._get_logo())
+
+    @api.model
+    def toggle_fold_onboarding_account_invoice(self):
+        """ Toggle the panel `folded` state. """
+        self.env.user.company_id.onboarding_account_invoice_folded =\
+            not self.env.user.company_id.onboarding_account_invoice_folded
+
+    @api.model
+    def close_onboarding_account_invoice(self):
+        """ Mark the panel as closed. """
+        self.env.user.company_id.onboarding_account_invoice_closed = True
+
+    @api.model
+    def onboarding_company(self, company_id):
+        """ Called by onboarding panel above the customer invoice list. """
+        action = self.env.ref('account.action_open_onboarding_company').read()[
+            0]
+        action['res_id'] = company_id
+        return action
+
+    @api.model
+    def onboarding_invoice(self, company_id):
+        """ Called by onboarding panel above the customer invoice list. """
+        action = self.env.ref('account.action_open_onboarding_invoice').read()[
+            0]
+        action['res_id'] = company_id
+        return action
+
+    def _get_sample_account(self, type_name, code_name):
+        """ If no account with code `code_name` exists, create a new one
+            with type `type_name` and code `code_name` and returns it """
+        sample_account = self.env['account.account'].search([('code', '=', code_name)], limit=1)
+
+        if len(sample_account) == 0:
+            sample_user_type_id = self.env['account.account.type'].search(
+                [('type', '=', type_name)], limit=1)
+            sample_account = self.env['account.account'].create({
+                'name': 'Sample Account ({})'.format(code_name),
+                'code': code_name,
+                'user_type_id': sample_user_type_id.id,
+                'reconcile': True,
+                'company_id': self.env.user.company_id.id,
+            })
+
+        return sample_account
+
+    def _get_sample_invoice(self):
+        """ Get a sample invoice or create it as well as a sample account,
+           journal, partner and line if it does not exist. """
+
+        sample_partner_name = 'Sample Partner'
+        sample_partner = self.env['res.partner'].search([('name', '=', sample_partner_name)], limit=1)
+
+        account_other_id = self._get_sample_account('other', 'SACOO').id
+        if len(sample_partner) == 0:
+            sample_partner = self.env['res.partner'].create({
+                'name': sample_partner_name,
+                'customer': True,
+                'email': self.env.user.email,
```

Comment:
By creating an new partner with the same email address than the current user, you will have side effects like : The user don't receive emails anymore ^^

Please use a sample email address.

Reply:
Having an administrator that can still receive emails is more important IMO.

Why don't we search a partner with the same email address than the user before trying to create a new one ?